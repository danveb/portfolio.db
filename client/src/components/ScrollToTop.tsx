import { useEffect } from "react";
import { useLocation } from "react-router-dom";

// NOTE: 
// when testing App it will definitely throw an error, but tests pass

export default function ScrollToTop() {
  const { pathname } = useLocation(); 
  useEffect(() => {
    window.scrollTo(0, 0); 
  }, [pathname]);
  return null; 
}; 