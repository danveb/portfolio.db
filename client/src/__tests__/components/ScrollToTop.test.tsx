import { render } from "@testing-library/react"; 
import ScrollToTop from "../../components/ScrollToTop"; 
import { BrowserRouter } from "react-router-dom";

describe("ScrollToTop component", () => {
  // smoke test
  test("renders without crashing", () => {
    render(
      <BrowserRouter>
        <ScrollToTop />
      </BrowserRouter>
    ); 
  }); 
  
  // snapshot test
  test("matches snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <ScrollToTop />
      </BrowserRouter>
    ); 
    expect(asFragment()).toMatchSnapshot(); 
  });
});