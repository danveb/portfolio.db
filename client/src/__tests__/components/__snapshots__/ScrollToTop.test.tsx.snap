// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTop component matches snapshot 1`] = `<DocumentFragment />`;
